proxy/src/main/java/com/wavefront/agent/preprocessor/predicate/ComparisonPredicate.java

```diff
@@ -9,10 +14,13 @@
  */
 public abstract class ComparisonPredicate<T> implements Predicate<T> {
   protected final String scope;
-  protected final String value;
+  protected final List<String> value;
 
-  public ComparisonPredicate(String scope, String value) {
+  public ComparisonPredicate(String scope, Object value) {
     this.scope = scope;
-    this.value = value;
+    Preconditions.checkArgument(value instanceof List || value instanceof String,
+        "Argument [value] should be of type [string] or [list].");
+    this.value =  (value instanceof List) ? ((List) value) :
+        Collections.unmodifiableList(Arrays.asList((String) value));
```

Comment:
I know premature optimization is the root of all evil, but I'm still curious about performance implications of wrapping a single string with a List and using List.stream(), especially considering that most real-life use cases would probably have just one value. I would expect using stream() with a single-element list would be about 10-15 times slower than simply looping through the collection. Won't be as elegant, of course. 
We can leave this for later iterations; LGTM for now.